soco/data_structures_entry.py

```diff
@@ -50,8 +50,20 @@ def from_didl_string(string):
             except KeyError:
                 raise DIDLMetadataError("Unknown UPnP class: %s" % item_class)
             item = cls.from_element(elt)
-            item = attempt_datastructure_upgrade(item)
-            items.append(item)
+            upgraded_item = attempt_datastructure_upgrade(item)
+
+            # Until music service items are able to get their own metadata from
+            # the music services, make it possible to copy it over when we have
+            # some e.g. when retrieving the Queue
+            # THIS SHOULD BE REMOVE WHEN MUSIC SERVICE ITEMS AND MUSIC SERVICES
+            # ARE FIXED
+            if music_service_medata_hack and isinstance(upgraded_item,
```

Comment:
Why not move this into attempt_datastructure_upgrade()?

Reply:
Good idea.